packages/material-ui/src/Menu/Menu.js

```diff
@@ -71,14 +109,31 @@ const Menu = React.forwardRef(function Menu(props, ref) {
     }
   };
 
+  const handleEntered = (element, isAppearing) => {
+    if (atLeastOneNestedMenu) setEntering(false);
+
+    if (onEntered) {
+      onEntered(element, isAppearing)
+    }
+  };
+  
   const handleListKeyDown = (event) => {
-    if (event.key === 'Tab') {
+    if (event.key === 'Tab' || event.key === 'Escape') {
       event.preventDefault();
+      setLastEnteredItemIndex(null);
 
       if (onClose) {
-        onClose(event, 'tabKeyDown');
+        onClose(event, `${event.key.toLowerCase()}KeyDown`);
       }
     }
+
+    if (event.key === 'ArrowLeft' && nestedMenu) {
+      // Tell the parent Menu to close the nested Menu that you're in, but
+      // don't trigger the nested Menu onClose cascade.
+      event.stopPropagation();
+      event.preventDefault();
```

Comment:
Do we need preventDefault here? Whats the default behavior for ArrowLeft that we're trying to block?

Reply:
Left scrolling. It's not common, but it's possible that someone might have a situation where when they ArrowLeft they both navigate up the menu stack and scroll left by some amount.